### ISSUE TITLE
Azure DNS label mismatch leads to endless failed attempts to ensure load balancer

### ISSUE THREAD

**Issue by novaterata** [kubernetes/kubernetes#63155]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

Ensuring Load Balancer times out and repeats endlessly

**What you expected to happen**:

And error of some kind that will prevent helm from waiting until timeout

**How to reproduce it (as minimally and precisely as possible)**:

Create an azure cluster with `--dns-label-prefix foo`

Create a kubernetes deployment with a service with the annotation:
`service.beta.kubernetes.io/azure-dns-label-name: bar`
Relevant PR:
https://github.com/kubernetes/kubernetes/pull/47849

**Anything else we need to know?**:

So basically this issue occurs when there is a mismatch between the dns label prefix upon cluster creation and the value used with the annotation

**Environment**:
- Kubernetes version (use `kubectl version`): 1.9.6
- Cloud provider or hardware configuration: Azure Container Services (AKS)
- Install tools: helm

/sig azure
/sig network

**Comment by khenidak** [comment]
@novaterata -- The service DNS label is for IP FE config on nodes LB. Not related to cluster DNS. Even for none `aks` configuration (acs-engine). Can you elaborate?

**Comment by novaterata** [comment]
I don't understand that sentence. All I know is that if they match, I get a
working FQDN for my load balancer. If they don't match then I get endless
timeouts

**Comment by feiskyer** [comment]
>  If they don't match then I get endless timeouts.

This is expected behavior. If there's something wrong when provisioning load balancer, controller manager will emit the warning event and then retry again.

**Comment by feiskyer** [comment]
> And error of some kind that will prevent helm from waiting until timeout

@novaterata Checking service's events is a working way, but I suspect a timeout is still required because the provision may success on retry.

**Comment by novaterata** [comment]
"controller manager will emit the warning event"
How would one see this event? It wasn't in the service events on the
dashboard where I saw the LB timeout occurring. The only warning was the
timeout. A log entry that mentions the azure dns label mismatch would be
adequate to close this issue for me.

**Comment by feiskyer** [comment]
In service events, e.g. `kubectl describe service <service-name>`. 

> A log entry that mentions the azure dns label mismatch would be adequate to close this issue for me.

I think there's already a log entry, but that's not user fencing (you can't expect helm to depend on it).

**Comment by justaugustus** [comment]
What's the status of this?
/assign @feiskyer

**Comment by feiskyer** [comment]
@justaugustus This is working as expected.

/close

**Comment by novaterata** [comment]
@feiskyer Can you please explain how to diagnose this in the logs? When I get an error saying the load balancer timed out waiting for the condition. How do I find out what that condition was? I just had this happen to me again, but this time the dns names did match. I upgraded from 1.9.6 to 1.11.2, so maybe something changed. I just want to know where are the logs for this? That was the main issue with this, that the Events in the dashboard do not provide helpful information. Please avoid too many acronyms and  jargon. I'm a fairly technical person, but all the replies to me so far on this issue have been unparsable.

**Comment by feiskyer** [comment]
@novaterata Please follow https://docs.microsoft.com/en-us/azure/aks/view-master-logs to check kube-controller-manager logs, which includes the detailed errors.

**Comment by novaterata** [comment]
@feiskyer I enabled diagnostics as instructed in your link, but there are no logs for kube-controller-manager

**Comment by feiskyer** [comment]
@novaterata It should show the logs after about 5-10 mins. If you still can't get the logs, please open a ticket on Azure portal.
